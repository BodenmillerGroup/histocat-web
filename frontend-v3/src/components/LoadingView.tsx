import { NonIdealState, Spinner } from "@blueprintjs/core";
import styles from "./LoadingView.module.scss";

type LoadingViewProps = {
  title?: string;
  description?: string;
};

export function LoadingView(props: LoadingViewProps) {
  const { title = "Loading...", description = "Please wait while data is loading" } = props;
  return (
    <div className={styles.container}>
      <NonIdealState title={title} icon={<Spinner />} description={description} />
    </div>
  );
}
