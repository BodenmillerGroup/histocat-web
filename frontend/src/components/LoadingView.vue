<template>
  <v-container fluid>
    <v-row>
      <v-col align="center" justify="center">
        <v-progress-circular
          indeterminate
          color="primary"
        />
        <h5>{{text}}</h5>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
  import { Component, Prop, Vue } from 'vue-property-decorator';

  @Component
  export default class LoadingView extends Vue {
    @Prop(String) text: string | undefined;
  }
</script>
