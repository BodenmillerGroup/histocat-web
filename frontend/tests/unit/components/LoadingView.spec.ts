import { mount } from "@vue/test-utils";
import LoadingView from "@/components/LoadingView.vue";

import Vue from "vue";
import Vuetify from "vuetify";
Vue.use(Vuetify);

describe("LoadingView.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(LoadingView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
