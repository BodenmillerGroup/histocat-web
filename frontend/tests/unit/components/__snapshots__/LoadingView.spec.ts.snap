// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingView.vue renders correctly 1`] = `
<div
  class="container container--fluid"
>
  <div
    class="row"
  >
    <div
      align="center"
      class="col"
      justify="center"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate primary--text"
        role="progressbar"
        style="height: 32px; width: 32px;"
      >
        <svg
          style="transform: rotate(0deg);"
          viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="v-progress-circular__overlay"
            cx="45.714285714285715"
            cy="45.714285714285715"
            fill="transparent"
            r="20"
            stroke-dasharray="125.664"
            stroke-dashoffset="125.66370614359172px"
            stroke-width="5.714285714285714"
          />
        </svg>
        <div
          class="v-progress-circular__info"
        />
      </div>
       
      <h5>
        
      </h5>
    </div>
  </div>
</div>
`;
